### ISSUE TITLE
Pangeo GCP Cost Breakdown

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo#615]
We have had some enhanced logging enabled for our Pangeo GCP billing over the past month or so. Here are some breakdowns:
https://gist.github.com/rabernat/21f97865129a97ca561f8448cd88b19a

### Overall Costs

![image](https://user-images.githubusercontent.com/1197350/57157638-47093180-6daf-11e9-8e25-f3bb5e2afea6.png)

### Compute Engine Costs by Cluster

![image](https://user-images.githubusercontent.com/1197350/57157653-538d8a00-6daf-11e9-9252-80bf25b833ac.png)

### Compute Engine Breakdown for Top Three Clusters

![image](https://user-images.githubusercontent.com/1197350/57157660-58523e00-6daf-11e9-96cf-debaa35d4c03.png)

![image](https://user-images.githubusercontent.com/1197350/57157663-5be5c500-6daf-11e9-9ce1-d82b79694d9c.png)

![image](https://user-images.githubusercontent.com/1197350/57157673-5ee0b580-6daf-11e9-992a-6d310e7f3fb8.png)

Bottom line, our burn rate of >$100 / day is unsustainable. We managed to get the costs down a bit recently, but in the past few weeks they have jumped up again. I attribute this to people doing actual big data science using lots of dask nodes on our production clusters. I'm a bit unclear, however, why we aren't taking more advantage of preemptible nodes for this.

Suggestions for how to dig deeper?

**Comment by yuvipanda** [comment]
@rabernat https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-usage-metering is going to help you dig much much deeper!

**Comment by jhamman** [comment]
I turned on cluster-usage-metering for both our dev-pangeo-io and binder clusters.

We aren't using preemptible nodes enough because we don't really know how to use dask-kubernetes taints/tolerations feature. If anyone has experience in this area, I could use a little help.

**Comment by TomAugspurger** [comment]
@jhamman I did this for a dask tutorial.  https://github.com/dask/dask-tutorial-infrastructure/blob/master/Makefile has at least part of it. Thats probably missing pieces, and is from a year or two ago, but may be helpful.

**Comment by TomAugspurger** [comment]
Oh, I wrote a readme (yay). Pointed me to the worker template too: https://github.com/dask/dask-tutorial-infrastructure/blob/master/notebook/worker-template.yaml

**Comment by jhamman** [comment]
Thanks @TomAugspurger. This is really helpful.

**Comment by jhamman** [comment]
Cross posting the blog post I published today on this subject: https://medium.com/pangeo/pangeo-cloud-costs-part1-f89842da411d
